istresearch/traptor
# Pulse 3957

This PR adds a call to `deactivate()` on stats collectors during shutdown to collapse time spent waiting for threads to exit.

**Testing**

Get the code

- `git clone git@github.com:istresearch/traptor.git`
- `git checkout PULSE-3957`

Set up your environment

- Create and activate a virtual environment (venv or Anaconda)
- Install the requirements: `pip install -r requirements.txt`

Run the tests

- In the root directory: `python -m pytest -vv --cache-clear --cov=traptor --cov-report html`

All tests should pass.

**Viewing DogWhistle Stats**

```
docker-compose build
docker-compose up -d
```

Visit Graphite at http://localhost:8000/

In the Metrics tree, under `stats/traptor`, a variety of metrics should be visible, eg, `starting_heartbeat`. Click on a stat to add it to th graph.  Change the graphite time window to something short (eg 1 hour) to see moving metrics.
